hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -1117,12 +1117,22 @@ public AbfsRestOperation read(final String path,
     return op;
   }
 
-  public AbfsRestOperation deletePath(final String path, final boolean recursive, final String continuation,
+  public AbfsRestOperation deletePath(final String path, final boolean recursive,
+                                      final String continuation,
                                       TracingContext tracingContext)
           throws AzureBlobFileSystemException {
-    final List<AbfsHttpHeader> requestHeaders = createDefaultHeaders();
-
+    final List<AbfsHttpHeader> requestHeaders
+        = (isPaginatedDelete(tracingContext, recursive)
+        && xMsVersion.compareTo(ApiVersion.AUG_03_2023) < 0)
+        ? createDefaultHeaders(ApiVersion.AUG_03_2023)
+        : createDefaultHeaders(xMsVersion);
     final AbfsUriQueryBuilder abfsUriQueryBuilder = createDefaultUriQueryBuilder();
+
+    if (isPaginatedDelete(tracingContext, recursive)) {
```

Comment:
what if this is a paginated delete but the L1126 api version test doesn't hold? Is that ok?

Reply:
If this is a paginated delete, then API version change condition will only fail if current version is greater than AUG_03_2023. 

In that case we can go ahead with Current API Version only as Azure APIs are backward compatible,